CMakeLists.txt

```diff
@@ -1,178 +1,152 @@
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#=============================================================================
-cmake_minimum_required(VERSION 3.17...3.18 FATAL_ERROR)
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
 
-project(RMM VERSION 0.17.0 LANGUAGES C CXX CUDA)
+project(
+  RMM
+  VERSION 0.17.0
+  LANGUAGES C CXX CUDA)
 
-include(cmake/CPM.cmake)
+include(cmake/Modules/CPM.cmake)
+include(cmake/Modules/RMM_thirdparty.cmake)
 
-###################################################################################################
-# - build type
+# build type
 
 # Set a default build type if none was specified
 set(DEFAULT_BUILD_TYPE "Release")
 
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
-  message(STATUS "Setting build type to '${DEFAULT_BUILD_TYPE}' since none specified.")
-  set(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}" CACHE
-      STRING "Choose the type of build." FORCE)
+  message(STATUS "RMM: Setting build type to '${DEFAULT_BUILD_TYPE}' since none specified.")
+  set(CMAKE_BUILD_TYPE
+      "${DEFAULT_BUILD_TYPE}"
+      CACHE STRING "Choose the type of build." FORCE)
   # Set the possible values of build type for cmake-gui
-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
-    "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel"
+                                               "RelWithDebInfo")
 endif()
 
-###################################################################################################
-# - compiler options
-
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-error=deprecated-declarations")
-endif(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-
-###################################################################################################
-# - build options
+# build options
 
 option(BUILD_TESTS "Configure CMake to build tests" ON)
 option(BUILD_BENCHMARKS "Configure CMake to build (google) benchmarks" OFF)
-
 option(DISABLE_DEPRECATION_WARNING "Disable warnings generated from deprecated declarations." OFF)
-if(DISABLE_DEPRECATION_WARNING)
-  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xcompiler -Wno-deprecated-declarations")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
-endif(DISABLE_DEPRECATION_WARNING)
+option(PER_THREAD_DEFAULT_STREAM "Build with per-thread default stream" OFF)
```

Comment:
There's a pending PR #603 which moves this option into just the benchmarks, which I think is a good idea.